src/theory/bags/bags_rewriter.cpp

```diff
@@ -0,0 +1,456 @@
+/*********************                                                        */
+/*! \file bags_rewriter.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mudathir Mohamed
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Bags theory rewriter.
+ **/
+
+#include "theory/bags/bags_rewriter.h"
+
+#include "normal_form.h"
+
+using namespace CVC4::kind;
+
+namespace CVC4 {
+namespace theory {
+namespace bags {
+
+BagsRewriteResponse::BagsRewriteResponse()
+    : d_status(REWRITE_DONE),
+      d_node(Node::null()),
+      d_rewrite(Rewrite::NO_REWRITE)
+{
+}
+
+BagsRewriteResponse::BagsRewriteResponse(RewriteStatus status,
+                                         Node n,
+                                         Rewrite rewrite)
+    : d_status(status), d_node(n), d_rewrite(rewrite)
+{
+}
+
+BagsRewriteResponse::BagsRewriteResponse(const BagsRewriteResponse& r)
+    : d_status(r.d_status), d_node(r.d_node), d_rewrite(r.d_rewrite)
+{
+}
+
+RewriteResponse BagsRewriteResponse::getResponse() const
+{
+  return RewriteResponse(d_status, d_node);
+}
+
+BagsRewriter::BagsRewriter(HistogramStat<Rewrite>* statistics)
+    : d_statistics(statistics)
+{
+  d_nm = NodeManager::currentNM();
+}
+
+RewriteResponse BagsRewriter::postRewrite(TNode n)
+{
+  BagsRewriteResponse response;
+  if (n.isConst())
+  {
+    // no need to rewrite n if it is already in a normal form
+    response = BagsRewriteResponse(REWRITE_DONE, n, Rewrite::NO_REWRITE);
+  }
+  else if (NormalForm::AreChildrenConstants(n))
+  {
+    Node normal = NormalForm::getNormalForm(n);
```

Comment:
I think the better name for this method is `NormalForm::evaluate`, whereas `getNormalForm` suggests something more general.  I believe we need a method whose sole purpose is to evaluate bag operators applied to constant arguments (which FYI will later need to be incorporated into https://github.com/CVC4/CVC4/blob/master/src/theory/evaluator.h when doing fast SyGuS involving bags).

Reply:
Done.